subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/rules/VariantFilesMetadataRulesIntegrationTest.groovy

```diff
@@ -507,6 +510,94 @@ class VariantFilesMetadataRulesIntegrationTest extends AbstractModuleDependencyR
         'java-runtime' | 'runtimeElements'
     }
 
+    @RequiredFeatures([
+        @RequiredFeature(feature = GradleMetadataResolveRunner.REPOSITORY_TYPE, value = "ivy"),
+        @RequiredFeature(feature = GradleMetadataResolveRunner.GRADLE_METADATA, value = "false")
+    ])
+    @Unroll
+    def "can add variants for ivy - #usageAttribute - honors conf based excludes "() {
+        // through this, we opt-into variant aware dependency management for a pure ivy module
+        given:
+        repository {
+            'org.test:moduleA:1.0' {
+                dependsOn 'org.test:moduleB:1.0'
+                excludeFromConfig 'org.test:moduleC', 'compile'
+                excludeFromConfig 'org.test:moduleC', 'runtime'
```

Comment:
I think the test might be richer if it showed that one branch excludes, while the other does not.
It would mean only a couple `if`s in the expected repo interactions and graph check.

Reply:
good call, I changed the test to add a `moduleD` and make sure that if it comes as part of A, it is excluded in the graph but if it is requested from another branch, it's just fine